src/dune/file_tree.ml

```diff
@@ -63,20 +62,30 @@ module Dune_file = struct
     | Plain x -> x.path
     | Ocaml_script p -> p
 
-  let load file ~project =
-    Io.with_lexbuf_from_file (Path.source file) ~f:(fun lb ->
-        if Dune_lexer.is_script lb then
-          Ocaml_script file
-        else
-          let sexps = Dune_lang.Parser.parse lb ~mode:Many in
-          let decoder =
-            Dune_project.set_parsing_context project Sub_dirs.decode
-          in
-          let sub_dirs, sexps =
-            Dune_lang.Decoder.parse decoder Univ_map.empty
-              (Dune_lang.Ast.List (Loc.none, sexps))
-          in
-          Plain { path = file; sexps; sub_dirs })
+  let load_plain file sexps ~parent ~project =
```

Comment:
Assuming `parent` represents the dune files *here* implied by the parent, I would call it `from_parent`. To me the name `parent` hints that we need to "descend" down from it, which seems to be not the case.

Reply:
I renamed it from_parent